### ISSUE TITLE
Tenant aware storage path in configuration file.

### ISSUE THREAD

**Issue by izshreyansh** [archtechx/tenancy#422]
_TL;DR; Is it possible to have tenant aware storage paths in configuration file?_

### Problem: Integration with Laravel Scout TNTSearch Driver

[TNT Search](https://github.com/teamtnt/laravel-scout-tntsearch-driver) is a full-text search driver for laravel that extends laravel's first party search driver: [Laravel Scout](https://laravel.com/docs/7.x/scout).

It basically creates `.index` file in filesystem based on its configuration option like this: 
```
'tntsearch' => [
    'storage'  => storage_path(),
    'asYouType' => true
]
```

So whenever it creates a new index file, It goes in `laravel\storage\users.index`. 
Instead of `laravel\storage\tenant46a4edef-7593-4a0d-961c-403473ea5117\users.index`.

Can someone please help on how can i provide scout a dynamic tenant aware storage path?

**Comment by izshreyansh** [comment]
I think its possible by making use of [Tenancy Bootstrappers](https://tenancyforlaravel.com/docs/v2/tenancy-bootstrappers/#tenancy-bootstrappers).

However i think there should be an inbuilt mechanism to replace `storage_path` to tenant's `storage_path`.  So its possible to achieve full isolation for tenant. Eg: `storage/logs`, `storage/framework` etc. 

@stancl Let me know what you think?

1.  Created bootstrapper in `App\TenancyBootstrappers\TntSearchTenancyBootstrapper`
```
class TntSearchTenancyBootstrapper implements TenancyBootstrapper
{
    public function start(Tenant $tenant)
    {
        config(['scout.tntsearch.storage' => storage_path() ]);
    }

    public function end()
    {
        config(['scout.tntsearch.storage' => storage_path() ]);
    }
}
```
It will switch out the default storage path set by NonTenantAware part of our application.

2. Register the bootstrapper in `config/tenancy` file
```
'bootstrappers' => [
    'scout' => App\TenancyBootstrappers\TntSearchTenancyBootstrapper::class,
]
```

**Comment by stancl** [comment]
Yeah, for this stuff I generally tell people to use event hooks & the `bootstrapping` event, since that doesn't require a custom class, but using a dedicated bootstrapper is cleaner in this case.

**Comment by renege** [comment]
@izshreyansh i tried this solution without any luck. error: unable to open database file.

**Comment by mindfullsilence** [comment]
Was able to get this working with the following. First by ensuring to implement the proper TenancyBootstraper interface (I'm using tenancy 3.x, so I think the above may have been for an older version), and by ensuring the directory exists. 

```php
<?php

namespace App\TenancyBootstrappers;

use Stancl\Tenancy\Contracts\TenancyBootstrapper;
use Stancl\Tenancy\Contracts\Tenant;

class TntSearchTenancyBootstrapper implements TenancyBootstrapper
{
    public function bootstrap(Tenant $tenant)
    {
        // looks something like /storage/tenant12e234-234e-48ae-957c-19813274593def
        $storagePath = storage_path();

        // This part probably belongs as a listener on the TenantCreated event,
        // which would also be a good place to run the scout:import command
        if(!is_dir($storagePath)) {
            mkdir($storagePath, 755, true);
        }

        config(['scout.tntsearch.storage' => $storagePath]);
    }

    public function revert()
    {
        config(['scout.tntsearch.storage' => storage_path()]);
    }
}
```

Then for config/tenancy.php:

```php
//...
'bootstrappers' => [
    //...
    App\TenancyBootstrappers\TntSearchTenancyBootstrapper::class,
],
//...
```

Creating the above, and then importing my scout models via command worked:

```sh
php artisan tenants:run scout:import --argument="model=App\\Models\\Tenant\\Taxonomy"
```
